Answers.md

```diff
@@ -2,10 +2,34 @@
+
 1.  What does it mean to _think_ in react?
 
+I don't know... My best answer to what I think this question is asking is that:
+
+React is built both with a distinct top-down heirarchy and with components. This impacts the way a React developer should think about approaching a project. 
+
+With components, the developer should endeavor to make each component do a single thing. Then, that component (and, likely, other components, too) is used by another component to do a bigger thing in concert. That means the developer should break down complex UIs into small parts.
+
+With top-down heirarchy, I am referring to the behind-the-scenes stuff that makes the reactivity of React work. In most cases, data and information is both retrieved and stored at a higher level than the bottom-level components. That data and information is then distributed to those lower-level components, and cannot be directly sent back up the chain (there are ways, of course, but it typically involves sending a function *from* the higher level along with the data), so React developers must be mindful of the heirarchy.
+
 1.  Briefly describe some of the differences between a Class/Stateful component and a Functional/Presentational component.
 
+A Class component has access to state, while a Functional component does not.
+
+A Class component is, perhaps, like a military officer, and a Functional component is like an enlisted soldier. Typically, a Class component creates many Functional components, just like a a military officer is typically commanding multiple enlisted soldiers. The officer's responsibility is to be mindful of what each of her soldiers is doing, and to coordinate their efforts toward mission success. She gives each of them orders, and they go actually carry them out. That's not to say the enlisted soldiers are mindless – in fact, Functional components can have complex logic of their own and can carry out very important tasks, but all they have are the orders they're given and they work with that. 
+
+Now, sometimes the officer gives soldiers a radio so they can report back in with new information. The soldiers cannot do anything about that information themselves, but they can report back. The officer has the new information, and can decide if anything needs to change. If it does, the officer can amend the orders to any or all of the soldiers as necessary, and the process continues.
+
 1.  Describe state.
 
+I like my analogy above. State is the commanding officer. Her job is to keep in mind all of the operation's moving parts. As the soldiers report back in, things change, and she knows about all of it. Then, as those things change, she amends the orders to the soldiers as needed.
+
+State is like a variable, held at a higher level in the React application. State is something that changes, typically when a user interacts with the web page. If the thing in question can be derived another way, it typically isn't state.
+

```

Comment:
This soldier comparison is gold! I think it makes a lot of sense! Great way to think about this!